tests/unit/test_project.py

```diff
@@ -33,6 +33,25 @@ def test_project_add_input_dataset(self):
         input_datasets = project.client.get(alias).successful().json()
         self.assertEqual(self.dataset_json, input_datasets)
 
+    @responses.activate
+    def test_project_remove_input_dataset(self):
+        dataset_id = self.dataset_json[0]["relativeId"]
+
+        responses.add(responses.GET, self.input_datasets_url, json=self.dataset_json)
+        responses.add(
+            responses.DELETE, f"{self.input_datasets_url}?id={dataset_id}", status=204
+        )
+        responses.add(responses.GET, self.input_datasets_url, json=[])
+
+        project = Project(self.unify, self.project_json[0])
+        dataset = list(project.input_datasets())[0]
```

Comment:
nit: to get just 1 item from a generator, you can use `next()`:

```suggestion
        dataset = next(project.input_datasets())
```